engine/chain/internal/bhs/verify_mr.go

```diff
@@ -0,0 +1,76 @@
+package bhs
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+
+	"github.com/bitcoin-sv/go-paymail/spv"
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/errors"
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/models"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+)
+
+/**
+NOTE: When you switch from enabled use_beef to disabled and restart the server,
+sometimes paymail capabilities are cached and VerifyMerkleRoots is called even though it shouldn't be.
+*/
+
+// VerifyMerkleRoots verifies the merkle roots of the given transactions using BHS request
+func (s *Service) VerifyMerkleRoots(ctx context.Context, merkleRoots []*spv.MerkleRootConfirmationRequestItem) (valid bool, err error) {
+	confirmations, err := s.mrVerifyRequest(ctx, merkleRoots)
+	if err != nil {
+		return false, err
+	}
+
+	switch confirmations.ConfirmationState {
+	case chainmodels.MRConfirmed:
+		return true, nil
+	case chainmodels.MRUnableToVerify:
+		s.logger.Warn().Msg("BHS is up but could not verify some merkle root(s). Defaulting to treat the provided merkle roots as valid.")
+		return true, nil
+	case chainmodels.MRInvalid:
+		return false, nil
+	default:
+		return false, spverrors.ErrInternal.Wrap(spverrors.Newf("unexpected confirmation state"))
+	}
+}
+
+func (s *Service) mrVerifyRequest(ctx context.Context, merkleRoots []*spv.MerkleRootConfirmationRequestItem) (*chainmodels.MerkleRootsConfirmations, error) {
```

Comment:
some more clear name please 🙏 
`requestForVerificationOfMerkleRoots`
`callToVerifyMerkleRoots`
`makeVerifyMerkleRootsRequest`

Reply:
Changed to `makeVerifyMerkleRootsRequest`.